source/usage-examples/find.txt

```diff
@@ -0,0 +1,164 @@
+lower ratings:
+
+.. code-block:: javascript
+
+   const sort = { rating: -1 }
+
+In this case, the number ``-1`` tells ``find`` to sort the fruit in
+descending order by rating, so the highest rated fruit comes first and
+the lowest rated fruit comes last. ``find`` returns the following
+documents when this sort is used with no query:
+
+.. code-block:: javascript
+
+   options = {};
+   options.sort = sort;
+   const cursor = collection.find({}, options);
+
+   { "_id": 4, "name": "avocados", "qty": 3, "rating": 5 }
+   { "_id": 1, "name": "apples", "qty": 5, "rating": 3 }
+   { "_id": 3, "name": "oranges", "qty": 6, "rating": 2 }
+   { "_id": 2, "name": "bananas", "qty": 7, "rating": 1 }
+
+
+Rewriting this sort to use a value of
+``1`` instead of a ``-1``, will sort our fruit in *ascending* order
+by rating, so the lowest rated fruit would come first. If you do not
+specify a sort order, the order of documents returned is
+`not guaranteed <https://docs.mongodb.com/manual/reference/method/cursor.sort/#result-ordering>`_.
+
+Projection
+~~~~~~~~~~
+
+Use a projection to control which fields ``find`` includes in the
+returned documents. Pass the following projection to only return the
+names of the fruits:
+
+.. code-block:: javascript
+
+   const projection = { name: 1 }
+
+``name`` has a value of ``1`` here to indicate that ``find`` should
+*include* the ``name`` field of each document returned. This implicitly
+*excludes* any fields that aren't explicitly included, so remember to
+specify any fields you want ``find`` to return with a ``1``. Passing
+this projection to ``find`` with no query and no sort order yields the
+following results:
+
+.. code-block:: javascript
+
+   options = {};
+   options.projection = projection;
+   const cursor = collection.find({}, options);
+
+   { "_id": 1, "name": "apples" }
+   { "_id": 2, "name": "bananas" }
+   { "_id": 3, "name": "oranges" }
+   { "_id": 4, "name": "avocados" }
+
+Despite the fact that this query only explicitly included the ``name``
+field, the ``_id`` field is returned as well! This happens because the
+``_id`` field is a unique identifier for each document, and if you
+want to refer to these documents again, you'd create a query document
+matching by ``_id`` in case there are duplicate ``name`` values in your
+collection. MongoDB anticipates that you might want a unique identifier
+for any documents you've found, so ``_id`` is always included unless you
+explicitly exclude it.
+
+You can explicitly *exclude* fields in your projections using a
+value of ``0``. This works just like include: if you explicitly exclude,
+all other fields are implicitly *included*.
+
+Return Value
+~~~~~~~~~~~~
+
+``find`` returns a
+`Cursor <https://mongodb.github.io/node-mongodb-native/3.3/api/Cursor.html>`_
+that contains the results of your query. If no document matches the
+query, the returned Cursor contains an empty collection of
+documents.
```

Comment:
**Return Value** is reference-oriented and doesn't really map to a use case. Maybe we could just mention this in the general description of how find works in the intro section?

Reply:
I could see readers misinterpreting "the cursor contains" to imply that the cursor object keeps the result set in memory. This isn't true - a cursor is a reference pointer to a documents in the result set, but doesn't itself contain the data.

You need to iterate through the cursor using a cursor method like `next()`, `toArray()`, or `forEach()` to actually fetch the documents.

Reply:
Agreed -- removed the "contains" reference, added notes about iteration to actually get contents, and moved to the intro section.